### PR TITLE
Site Settings: Update Traffic section to use the new settings controller

### PR DESCRIPTION
**Pull request by tyxla**
This PR updates the Traffic section to use the new settings controller that we introduced in #12493. Previously we were duplicating almost the same code for the Traffic section controller, and now we'll be using the same one for all other settings sections that we'll separate from the settings chunk. 

This PR adds no real performance improvement (as the Traffic tab is already a separate chunk), but it demonstrates how we can move a settings tab in its own section. Following this example, we can move out the rest of the settings tabs in their own chunks, which will reduce the size of the main `settings` one.

For background on why we're moving settings sections to separate chunks, you can refer to p4TIVU-5Zs-p2.

To test:
* Checkout this branch.
* Test all Traffic tab cases and verify there are no regressions there: 
  * `/settings/traffic/`
  * `/settings/traffic/$site` for a Jetpack site
  * `/settings/traffic/$site` for a WordPress.com site
* Verify there are no regressions in the settings and toggles in the Traffic tab by playing with them a bit.

### CODE REVIEW

File: client/my-sites/site-settings/traffic/controller.js [hunk 1]
```diff
@@ -7,50 +7,27 @@ import React from 'react';
 /**
```
Comment (mtias) [inline client/my-sites/site-settings/traffic/controller.js]: Unrelated to this PR, but we probably want to rename the folder to be `settings-traffic`.

Comment (tyxla) [inline client/my-sites/site-settings/traffic/controller.js]: Good call, but doesn't `settings` get redundant, considering the fact that it's contained in the path too (`site-settings/settings-traffic`)?

**Comment by dmsnell** [comment]
Not much opinion here - looks fine to me if it doesn't break anything - thanks for the work!

File: client/my-sites/site-settings/traffic/controller.js (hunk 1 above)
Comment (tyxla) [inline client/my-sites/site-settings/traffic/controller.js]: Disregard the above comment; renamed.

**Comment by tyxla** [comment]
@mtias I've addressed your feedback, this one could use another review.

**Comment by tyxla** [comment]
Just to clarify: tests are failing because `bin/run-lint` does not like that we're introducing a new usage of `sites-list`, but we're actually just moving that code. 

I intend to merge this one by creating a merge commit, which will trigger a new build, in which that error will not be considered.

**Comment by nosolosw** [comment]
The problem that Marin mentions is actually this: https://github.com/Automattic/eslines/issues/2 

Our linting process checks only for linting errors in new/modified lines. As `eslines` doesn't differentiate between moved and new files, it's going to check them all as new/modified lines.

**Comment by tyxla** [comment]
This one has been rebased again.

**Comment by tyxla** [comment]
Thanks for the review @sirreal! I'll wait for #13258 to land, as it intends to get rid of `sites-list` here.

**Comment by tyxla** [comment]
Will address in a new PR.

**Pull request closed without merging**